File: py/leetcode-parking.py
class ParkingSystem:
    
    def __init__(self, big: int, medium: int, small: int):
        self.bs = big
        self.ms = medium
        self.ss = small    

    def addCar(self, carType: int) -> bool:
        b = False
        if carType == 1:
            if self.bs > 0:
                b = True
                self.bs -= 1
        if carType == 2:
            if self.ms > 0:
                b = True
                self.ms -= 1
        if carType == 3:
            if self.ss > 0:
                b = True
                self.ss -= 1
        return b


# Your ParkingSystem object will be instantiated and called as such:
# obj = ParkingSystem(big, medium, small)
# param_1 = obj.addCar(carType)

File: rust/src/leetcode-parking.rs
struct ParkingSystem {
    Bs:i32,
    Ms:i32,
    Ss:i32
}


/** 
 * `&self` means the method takes an immutable reference.
 * If you need a mutable reference, change it to `&mut self` instead.
 */
impl ParkingSystem {

    fn new(big: i32, medium: i32, small: i32) -> Self {
        Self{
            Bs:big,
            Ms:medium,
            Ss:small
        }
    }
    
    fn add_car(&mut self, car_type: i32) -> bool {
        let mut b:bool = false;
        let mut slot = match car_type {
            1 => &mut self.Bs,
            2 => &mut self.Ms,
            3 => &mut self.Ss,
            _ => &mut self.Bs
        };

        if slot > &mut 0 {
            *slot -= 1;
            b = true;
        }
        b
    }
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * let obj = ParkingSystem::new(big, medium, small);
 * let ret_1: bool = obj.add_car(carType);
 */